app/abci.go

```diff
@@ -35,7 +38,38 @@ func (app *App) EndBlock(ctx sdk.Context, req abci.RequestEndBlock) (res abci.Re
 func (app *App) CheckTx(ctx context.Context, req *abci.RequestCheckTx) (*abci.ResponseCheckTxV2, error) {
 	_, span := app.GetBaseApp().TracingInfo.Start("CheckTx")
 	defer span.End()
-	return app.BaseApp.CheckTx(ctx, req)
+	resp, err := app.BaseApp.CheckTx(ctx, req)
+	if err != nil {
+		return resp, err
+	}
+
+	// For EVM, populate sender on the response so that mempool can decide to dedupe
+	tx, err := app.txDecoder(req.Tx)
```

Comment:
decoding (and the addr recovery logic below) is quite heavy so ideally we don't repeat it here. One other option is to a new field in `sdk.Context`, set it in the `sig` antehandler (where we derive address), and set it in checkTx response in sei-cosmos:baseapp:runTx.

But for now I think this is okay, since CheckTx itself doesn't directly affect chain performance